datasource/netcdf_groundwater.py

```diff
@@ -280,121 +277,115 @@ def get_gridadmin(self, variable=None):
+    #     result_filter = model.timeseries(indexes=time_slice)
+    #
+    #     if node_ids is None:
+    #         result_filter = result_filter.filter(id__gt=0)
+    #     result = result_filter.get_filtered_field_value(variable)
+    #
+    #     if node_ids is not None:
+    #         # Unfortunately h5py/threedigrid indexing is not as fancy as
+    #         # numpy, i.e. we can't use duplicate indexes/unsorted indexes.
+    #         # Thus we load a bit more in memory as a numpy array and then apply
+    #         # the final indexing with numpy.
+    #         return result[time_index_filter][node_ids]
+    #     else:
+    #         return result[time_index_filter]
+
     def get_values_by_timestep_nr(
-        self, variable, timestamp_idx, index=None, use_cache=True
-    ):
-        return self.temp_get_values_by_timestep_nr_impl(
-            variable, timestamp_idx, index, use_cache
-        )
+            self, variable, timestamp_idx, node_ids=None, use_cache=True):
+        """Return an array of values of the given variable on the specified timestamp(s)
 
-    def _nc_from_mem(self, ds, variable, use_cache=True):
-        """Get netcdf data from memory if needed."""
-        if use_cache:
-            try:
-                data = self._cache[variable]
-            except KeyError:
-                # Keep the whole netCDF array for a variable in memory for
-                # performance
-                data = ds.get(variable)[:]
-                self._cache[variable] = data
-        else:
-            # this returns a netCDF Variable, which behaves like a np array
-            data = ds.get(variable)
-        return data
+        If only one timestamp is specified, a 1d np.array is returned.  If an
+        array of multiple timestamp_idx is given, a 2d np.array is returned.
 
-    def temp_get_values_by_timestep_nr_impl(
-        self, variable, timestamp_idx, index=None, use_cache=True
-    ):
-        var_2d = self.PREFIX_2D + variable
-        var_1d = self.PREFIX_1D + variable
+        If node_ids is specified, only the node_ids specified in the nodes will
+        be returned.
 
-        # determine if it's an agg var
-        if variable in self.available_subgrid_map_vars:
-            ds = self.ds
-        elif variable in self.available_aggregation_vars:
-            ds = self.ds_aggregation
+        :param variable: (str) variable name, e.g. 's1', 'q_pump'
+        :param timestamp_idx: int or 1d numpy.array of indexes of timestamps
+        :param node_ids: 1d numpy.array of node_ids or None in which case all
+            nodes are returned.
+        :param use_cache: (bool)
+        :return: 1d/2d numpy.array
+        """
+        values = self._nc_from_mem(variable)
+        if isinstance(timestamp_idx, int):
+            timestamp_idx = np.array([timestamp_idx])
+
+        if node_ids is None:
+            # If no node_ids are specified, we don't want to return the first
```

Comment:
Ik zou verwachten dat in de "else" dan het andere is dat het eerste ding wèl wordt meegegeven ofzo. Maar er gebeurt iets anders. => commentaar is niet compleet of verwart een beetje of er moet ook commentaar bij de else?